src/DotNetty.Codecs.Redis/Messages/RedisCodecUtil.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace DotNetty.Codecs.Redis.Messages
+{
+    using System;
+    using System.Diagnostics.Contracts;
+    using System.Globalization;
+    using System.Text;
+    using DotNetty.Buffers;
+
+    static class RedisCodecUtil
+    {
+        internal const char RedisSimpleString = '+';
+        internal const char RedisError        = '-';
+        internal const char RedisInteger      = ':';
+        internal const char RedisBulkString   = '$';
+        internal const char RedisArray        = '*';
+
+        internal static byte[] GetBytes(this IRedisMessagePool messagePool, long value)
+        {
+            Contract.Requires(messagePool != null);
+
+            byte[] bytes;
+            if (!messagePool.TryGetBytes(value, out bytes))
+            {
+                bytes = LongToAsciiBytes(value);
+            }
+
+            return bytes;
+        }
+
+        internal static SimpleStringRedisMessage GetSimpleStringMessage(this IRedisMessagePool messagePool, IByteBuffer byteBuffer)
+        {
+            Contract.Requires(messagePool != null);
+            Contract.Requires(byteBuffer != null);
+
+            SimpleStringRedisMessage message;
+            if (!messagePool.TryGetMessage(byteBuffer, out message))
+            {
+                message = new SimpleStringRedisMessage(byteBuffer.ToString(Encoding.UTF8));
+            }
+
+            return message;
+        }
+
+        internal static ErrorRedisMessage GetErrorMessage(this IRedisMessagePool messagePool, IByteBuffer byteBuffer)
+        {
+            Contract.Requires(messagePool != null);
+            Contract.Requires(byteBuffer != null);
+
+            ErrorRedisMessage message;
+            if (!messagePool.TryGetMessage(byteBuffer, out message))
+            {
+                message = new ErrorRedisMessage(byteBuffer.ToString(Encoding.UTF8));
+            }
+
+            return message;
+        }
+
+        internal static bool IsInline(this RedisMessageType messageType) => 
+            messageType == RedisMessageType.SimpleString
+            || messageType == RedisMessageType.Error
+            || messageType == RedisMessageType.Integer;
+
+        internal static RedisMessageType GetMessageType(this byte byteCode)
```

Comment:
>this byte [](start = 56, length = 9)

don't declare extension method on common types unless the method is universally useful. E.g. I generally don't want to convert `byte` value to `RedisMessageType`. Just drop `this` and invoke it like `RedisCodecUtil.GetMessageType(byteCode)`.
Also, consider naming it `ParseMessageType`.